rapids-4-spark-tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ClassWarehouse.scala

```diff
@@ -44,4 +44,65 @@ case class ExecutorCase(
   executorID: String, host: String, totalCores: Int, resourceProfileId: Int)
 
 case class ExecutorRemovedCase(
-  executorID: String, reason: String, time: Long)
\ No newline at end of file
+  executorID: String, reason: String, time: Long)
+
+case class SQLExecutionCase(
+    sqlID: Long, description: String, details: String,
+    startTime: Long, endTime: Option[Long],
+    duration: Option[Long], durationStr: String)
+
+case class SQLPlanMetricsCase(
+    sqlID: Long, name: String, accumulatorId: Long,
+    metricType: String)
+
+case class PlanNodeAccumCase(
+    sqlID: Long, nodeID: Long, nodeName:String,
+    nodeDesc: String, accumulatorId: Long)
+
+case class DriverAccumCase(
+    sqlID: Long, accumulatorId: Long, value: Long)
+
+case class TaskStageAccumCase(
+    stageId: Int, attemptId: Int, taskId: Option[Long],
+    accumulatorId: Long, name: Option[String], value: Option[Long],
+    isInternal: Boolean)
+
+case class JobCase(
+    jobID: Int, stageIds: Seq[Int] , sqlID: Option[Long],
+    properties: scala.collection.Map[String, String], startTime: Long,
+    endTime: Option[Long], jobResult: Option[String], failedReason: String,
+    duration: Option[Long], durationStr: String, gpuMode: Boolean)
+
+case class StageCase(
+    stageId: Int, attemptId: Int , name: String,
+    numTasks: Int, numRDD: Int, parentIds: Seq[Int],
+    details: String, properties: scala.collection.Map[String, String],
+    submissionTime: Option[Long], completionTime: Option[Long],
+    failureReason: Option[String], duration: Option[Long],
+    durationStr: String, gpuMode: Boolean)
+
+// Note: sr = Shuffle Read; sw = Shuffle Write
+// Totally 39 columns
+case class TaskCase(
```

Comment:
are these all just copies of the spark events essentially?  I'm assuming we can't use those to create the data frames though as easily, correct?

Reply:
Yes, these are case classes which hold information for Task related events. Likewise, job/stage related events are stored in  JobCase and StageCase. Please let me know if you have better way to implement this. We can do it here or as follow on.